shpc/utils/fileio.py

```diff
@@ -114,12 +115,14 @@ def copyfile(source, destination, force=True):
     return destination
 
 
-def write_file(filename, content, mode="w"):
+def write_file(filename, content, mode="w", exec=False):
     """
     Write content to a filename
     """
     with open(filename, mode) as filey:
         filey.writelines(content)
+    if exec:
```

Comment:
Let's avoid using subprocess here - maybe try:

```bash
import os
import stat

st = os.stat(filename)
os.chmod(filename, st.st_mode | stat.S_IEXEC)
```

Reply:
added in commit https://github.com/marcodelapierre/singularity-hpc/commit/0d433265cba7d6700559105391ec15a64737e44d

Reply:
also tested